docs/_source/guides/llms/practical_guides/fine_tune.md

```diff
@@ -518,149 +796,111 @@ encoding = tokenizer([inputs], return_tensors="pt")
-samples, rewards = [], []
-for entry in dataset:
-    samples.append(entry["poorer_response"])
-    rewards.append(1)
-    samples.append(entry["better_response"])
-    rewards.append(2)
+from argilla.feedback import TrainingTask
+from typing import Dict, Any, Iterator
 
-trainer = trlx.train(config=config, samples=samples, rewards=rewards)
-```
+template = """\
+### Instruction: {instruction}\n
+### Context: {context}\n
+### Response: {response}"""
 
-:::
+def formatting_func(sample: Dict[str, Any]) -> Iterator[Tuple[str, str]]:
+    # Our annotators were asked to provide new responses, which we assume are better than the originals
+    og_instruction = sample["original-instruction"]
+    og_context = sample["original-context"]
+    rejected = sample["original-response"]
+    prompt = template.format(instruction=og_instruction, context=og_context, response="")
 
-::::
+    for instruction, context, response in zip(sample["new-instruction"], sample["new-context"], sample["new-response"]):
+        if response["status"] == "submitted":
+            chosen = response["value"]
+            if chosen != rejected:
+                yield prompt, chosen, rejected
 
-## Pre-training
 
-When talking about pre-training, we generally talk about a simple `prompt-completion` task, where we need the model to pick up on basic statistics of the language it is learning. Given that you are familiar with Spanish cuisine and the prompt sentence, `The base ingredient of paella is ___`, you know that the word in the `___` is much more likely to be `rice` than `apples`.  So, you are basically training a causal language model or text generation model.
-
-```{note}
-This is an unsupervised approach hence we only infer training data from a basic sentence like `The base ingredient of paella is rice.` by starting with the word `The`, and from there unwrapping the sentence step by step.
+task = TrainingTask.for_direct_preference_optimization(formatting_func=formatting_func)
 ```
 
-### Data
-
-Many training datasets for this task can be found online (e.g., [Hugging Face](https://huggingface.co/datasets?task_categories=task_categories:text-generation&sort=downloads)). You can either upload this in the right Argilla format but it might be needed to collect and fine-tune additional data with Argilla. So we, therefore, provide a basic setup underneath which should help you to start gathering or preparing pre-training data.
-
-```{note}
-When it comes to pre-training an LLM, we generally do not need data of highest quality, but it is always smart to use domain-specfic data and to avoid data that might lead to undecired effect like hallucination and bias.
-```
+**ArgillaTrainer**
 
-First, create a `FeedbackDataset` with records.
+Instead of using this dataset, we'll use the task directly with our `FeedbackDataset` in the `ArgillaTrainer`. In contrary to PPO, we do not need to specify any reward model, because this preference modeling is inferred internally by the DPO-algorithm.
```

Comment:
```suggestion
We'll use the task directly with our `FeedbackDataset` in the `ArgillaTrainer`. In contrary to PPO, we do not need to specify any reward model, because this preference modeling is inferred internally by the DPO-algorithm.
```
This was taken from somewhere that followed a snippet of how to observe this dataset. Without that, we don't need this "intro"